### PR TITLE
Strubiums weaponlib cleanup

### PR DESCRIPTION
**Pull request by strubium**
## 📝 Description

Cleanup the various things that annoy me in weaponlib

## 🎯 Goals

- Cleanup the naughty files in weaponlib
- Add all the changes to the Changelog

## ❌ Non Goals

- Break the mod, smeagle It even 

## 🚦 Testing 

👍 

## ⏮️ Backwards Compatibility 

1.12.2

## 📚 Related Issues & Documents

N/A

## 📖 Added to documentation?

- [ ] 📜 README.md
- [ ] 📑 Documentation
- [ ] 📓 Javadoc
- [ ] 🍕 Comments
- [x] 🙅 No documentation needed

### CODE REVIEW

File: src/main/java/com/paneedah/weaponlib/render/SpriteSheetTools.java [hunk 1]
```diff
@@ -43,38 +43,55 @@ public double getMaxV() {
...
-            return new Sprite(u, v, u + (spriteWidth / width), v + (spriteHeight / height));
-
+        // If flipping horizontally, swap minU and maxU
+        if (flip) {
+            double temp = minU;
+            minU = maxU;
+            maxU = temp;
         }
 
+        return new Sprite(minU, minV, maxU, maxV);
```
[the remark this answers is not in the document]
Comment (Desoroxxx) [reply src/main/java/com/paneedah/weaponlib/render/SpriteSheetTools.java:95]: Not now

**Comment by strubium** [comment]
I've decided to rename WorldGeneratorEventHandler to ModernWorldGenerator, because it isn't a event handler. I've also made an EquipmentFactory interface, it is like VehicleFactory and TileEntityFactory 

This PR also lowers the total number of Qodana problems from 21,962 to 20,866

File: src/main/java/com/paneedah/mwc/ModernWorldGenerator.java [hunk 2]
```diff
@@ -12,9 +12,9 @@
 
 import java.util.Random;
 
-public class WorldGeneratorEventHandler implements IWorldGenerator {
+public class ModernWorldGenerator implements IWorldGenerator {
 
-    public WorldGeneratorEventHandler() {
+    public ModernWorldGenerator() {
 
     }
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/ModernWorldGenerator.java]: Replace this with a Lombok anotation

File: src/main/java/com/paneedah/mwc/UniversalSoundRegistry.java [hunk 3]
```diff
@@ -5,13 +5,14 @@
 /**
  * Creates a sound registry for WeaponLib to access in order to play
  * reloading noises.
+ * <br> <br>
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/UniversalSoundRegistry.java]: Do not use br, simply put an empty line

File: src/main/java/com/paneedah/mwc/equipment/EquipmentFactory.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.paneedah.mwc.equipment;
+
+import com.paneedah.weaponlib.ModContext;
+
+public interface EquipmentFactory {
+
+    void createEquipment(ModContext modContext);
+
+}
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/equipment/EquipmentFactory.java]: No, we will just use new, but that will need me finishing the equipment rework

**Review by Desoroxxx — CHANGES_REQUESTED** [review]

File: src/main/java/com/paneedah/mwc/ModernWorldGenerator.java (hunk 2 above)
Comment (strubium) [reply src/main/java/com/paneedah/mwc/ModernWorldGenerator.java]: Done

File: src/main/java/com/paneedah/mwc/UniversalSoundRegistry.java (hunk 3 above)
Comment (strubium) [reply src/main/java/com/paneedah/mwc/UniversalSoundRegistry.java]: Done

File: src/main/java/com/paneedah/mwc/equipment/EquipmentFactory.java (hunk 4 above)
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/equipment/EquipmentFactory.java]: Remove modContext from this, to access a `ModContext` uses `MWC.modContext`

File: src/main/java/com/paneedah/mwc/proxies/CommonProxy.java [hunk 5]
```diff
@@ -62,20 +63,24 @@ public void preInit(final MWC mod) {
...
         SpecialAttachments.init(mod, MWC.modContext);
-        Backpacks.init();
-        Belts.init();
-        Vests.init(mod);
-        Armors.init(mod, MWC.modContext);
+
+        new Backpacks().createEquipment(MWC.modContext);
+        new Belts().createEquipment(MWC.modContext);
+        new Vests().createEquipment(MWC.modContext);
+        new Armors().createEquipment(MWC.modContext);
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/proxies/CommonProxy.java]: Make this static and not use new

File: src/main/java/com/paneedah/weaponlib/tile/LootBoxTileEntity.java [hunk 6]
```diff
@@ -35,7 +36,7 @@ public void dropContents(World world, BlockPos pos, EntityPlayer player) {
                 lastEquipmentDispenseTimestamp = world.getWorldTime();
                 Equipment equipment = configuration.getEquipmentOptions().pick(EnumDifficulty.EASY);
                 if (equipment != null && equipment.item != null) {
-                    System.out.println("Dropping " + equipment.item.getTranslationKey());
+                    ProjectConstants.LOGGER.info("Dropping " + equipment.item.getTranslationKey());
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/tile/LootBoxTileEntity.java]: Use static import

File: src/main/java/com/paneedah/weaponlib/tile/CustomTileEntityRenderer.java [hunk 7]
```diff
@@ -20,30 +31,52 @@ public CustomTileEntityRenderer(ModelBase model, ResourceLocation textureResourc
...
-
-        //System.out.println("Rendering side " + ((CustomTileEntity<?>) tileEntity).getSide());
-//        GL11.glRotatef(-90f * ((CustomTileEntity<?>) tileEntity).getSide(), 0, 1f, 0);
-        positioning.accept(tileEntity);
-        //GL11.glEnable(GL11.GL_CULL_FACE);
-        model.render(null, 0f, 0f, 0f, 0f, 0f, 0.0625f);
-
-        //  GL11.glDisable(GL12.GL_RESCALE_NORMAL);
-        GL11.glPopMatrix();
     }
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/tile/CustomTileEntityRenderer.java:81]: Please keeps comments for now, until I rewrite that part of rendering, it helps figure out what the hell was going on

File: src/main/java/com/paneedah/weaponlib/render/scopes/Reticle.java [hunk 8]
```diff
@@ -11,11 +13,13 @@ public class Reticle {
...
 
+    @Getter @Setter
     private ResourceLocation reticleTexture;
     private Type reticleType;
 
-
+    @Setter @Getter
     private float textureScale;
+    @Setter @Getter
     private Vec3d backgroundColor;
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/render/scopes/Reticle.java]: Please set the annotations on the same line

File: src/main/java/com/paneedah/weaponlib/animation/AnimationModeProcessor.java [hunk 9]
```diff
@@ -215,321 +215,147 @@ public Transform getTransformFromSelected() {
     }
 
     public void onTick() {
-
-
-        //this.transformMode = 1;
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/weaponlib/animation/AnimationModeProcessor.java:220]: Keep that one comment, could be useful

File: src/main/java/com/paneedah/mwc/proxies/CommonProxy.java [hunk 10]
```diff
@@ -62,20 +63,24 @@ public void preInit(final MWC mod) {
...
+
         Attachments.init(mod);
         AuxiliaryAttachments.init(mod);
         GunSkins.init(mod);
         Bullets.init(mod);
         Magazines.init(mod);
         Guns.init(mod, this);
-        Electronics.init();
+        new Electronics().createEquipment(MWC.modContext);
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/proxies/CommonProxy.java]: Make this static and not use new

File: src/main/java/com/paneedah/mwc/proxies/CommonProxy.java [hunk 11]
```diff
@@ -62,20 +63,24 @@ public void preInit(final MWC mod) {
...
         Grenades.init(mod, this);
         CustomSpawnEggs.init(mod, this);
-        TileEntities.init(this);
+
+        new TurretBaseFactory().createTileEntity(MWC.modContext);
+        new TileEntities().createTileEntity(MWC.modContext);
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/proxies/CommonProxy.java]: Make this static and not use new

File: src/main/java/com/paneedah/mwc/utils/PlayerCreatureWrapper.java [hunk 12]
```diff
@@ -18,7 +20,7 @@ public class PlayerCreatureWrapper extends EntityPlayerSP {
     private EntityLivingBase entityLiving;
 
     public PlayerCreatureWrapper(World world) {
-        super(MC, world, MC.getConnection(), new StatisticsManager(), new RecipeBookClient());
+        super(MC, world, Objects.requireNonNull(MC.getConnection()), new StatisticsManager(), new RecipeBookClient());
```
Comment (Desoroxxx) [inline src/main/java/com/paneedah/mwc/utils/PlayerCreatureWrapper.java]: This is useless

**Review by Desoroxxx — CHANGES_REQUESTED** [review]

File: src/main/java/com/paneedah/weaponlib/render/scopes/Reticle.java (hunk 8 above)
Comment (strubium) [reply src/main/java/com/paneedah/weaponlib/render/scopes/Reticle.java]: Done

File: src/main/java/com/paneedah/mwc/proxies/CommonProxy.java (hunk 5 above)
Comment (strubium) [reply src/main/java/com/paneedah/mwc/proxies/CommonProxy.java]: Done

File: src/main/java/com/paneedah/mwc/equipment/EquipmentFactory.java (hunk 4 above)
Comment (strubium) [reply src/main/java/com/paneedah/mwc/equipment/EquipmentFactory.java]: Done

File: src/main/java/com/paneedah/weaponlib/tile/LootBoxTileEntity.java (hunk 6 above)
Comment (strubium) [reply src/main/java/com/paneedah/weaponlib/tile/LootBoxTileEntity.java]: Done

File: src/main/java/com/paneedah/weaponlib/animation/AnimationModeProcessor.java (hunk 9 above)
Comment (strubium) [reply src/main/java/com/paneedah/weaponlib/animation/AnimationModeProcessor.java:220]: Done

File: src/main/java/com/paneedah/mwc/proxies/CommonProxy.java (hunk 10 above)
Comment (strubium) [reply src/main/java/com/paneedah/mwc/proxies/CommonProxy.java]: Done

File: src/main/java/com/paneedah/mwc/utils/PlayerCreatureWrapper.java (hunk 12 above)
Comment (strubium) [reply src/main/java/com/paneedah/mwc/utils/PlayerCreatureWrapper.java]: Done

File: src/main/java/com/paneedah/mwc/proxies/CommonProxy.java (hunk 11 above)
Comment (strubium) [reply src/main/java/com/paneedah/mwc/proxies/CommonProxy.java]: Done

**Review by Desoroxxx — APPROVED** [review]

**Pull request merged**